ArduCopter/autoyaw.cpp

```diff
@@ -19,22 +19,22 @@ float Mode::AutoYaw::look_ahead_yaw()
     return _look_ahead_yaw;
 }
 
-void Mode::AutoYaw::set_mode_to_default(bool rtl)
+void Mode::AutoYaw::set_mode_to_default(Mode::Number mode)
 {
-    set_mode(default_mode(rtl));
+    set_mode(default_mode(mode));
 }
 
 // default_mode - returns auto_yaw.mode() based on WP_YAW_BEHAVIOR parameter
 // set rtl parameter to true if this is during an RTL
-autopilot_yaw_mode Mode::AutoYaw::default_mode(bool rtl) const
+autopilot_yaw_mode Mode::AutoYaw::default_mode(Mode::Number mode) const
 {
     switch (copter.g.wp_yaw_behavior) {
 
     case WP_YAW_BEHAVIOR_NONE:
         return AUTO_YAW_HOLD;
 
     case WP_YAW_BEHAVIOR_LOOK_AT_NEXT_WP_EXCEPT_RTL:
-        if (rtl) {
+        if ((mode == Mode::Number::RTL) || (mode == Mode::Number::SMART_RTL)) {
```

Comment:
Please do not nominate mode numbers like this.  Ask the mode about itself, rather than assuming behaviour based on mode number.

Also pass the mode, not the mode number

Reply:
I implemented
`virtual bool is_rtl_mode() const { return false; }`
and returns true only in RTL & SMART_RTL.

I wonder if it is better to use 
`virtual Number mode_number() const;` rather than implementing a function for every mode.

what do you think ?